core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/JsonSerializer.java

```diff
@@ -105,9 +101,30 @@ private static void writeView(View view, JsonObject json) {
             // if we're using relations, constituents shouldn't have any duplicates; otherwise upon deserialization
             // things would not be as expected
             Set<Constituent> consSet = new HashSet<>(constituents);
-            if(consSet.size() < constituents.size())
-                logger.error("There are duplicate constituents in the '" + view + "' view. " +
-                        "You have to fix this otherwise things will be messed up, upon deserialization. ");
+            if(consSet.size() < constituents.size()) {
+                logger.error("There are "+(constituents.size()-consSet.size())+" duplicates constituents in the '" + view.getViewName() + "' view. " +
+                        "You have to fix this otherwise things will be messed up, upon deserialization : "+view.getTextAnnotation().getId());
+                // identify the constituents that hash the same, and print them.
+                int hits = 0;
+                for (int outter = 0 ; outter < constituents.size(); outter++) {
+                    for (int inner = 0 ; inner < constituents.size(); inner++) {
+                        if(outter == inner)
+                            continue;
+                        else {
+                            Constituent c1 = constituents.get(outter);
+                            Constituent c2 = constituents.get(inner);
+                            if (c1.equals(c2)) {
+                                int s1 = c1.getStartSpan();
+                                int s2 = c2.getStartSpan();
+                                c1.equals(c2);
+                                logger.error(hits+") Got them : "+c1+"("+s1+") -- "+c2+"("+s2+")");
```

Comment:
@cowchipkid this is the place you nitced wehave lots of hash colisions? #562

Reply:
Yes. We will need to come back to this. I wonder though if these collisions are the result of my poorly constructed constituents (I use using similar labels for both constituents and relations). I will need to come back to this, I want to see how big a problem this is on a less complete view, just tokens.

Reply:
Do we want to address this issue before merging the pr?

Reply:
I hate to leave this. But it's been like this forever as far as I can see, and it might take a while to really figure it out, so I would rather not delay this PR. I have a rare opportunity to use this huge machine to do the param sweep, offer ends at the end of next week, so I need to get on it. I think I will create another branch to work on this issue along with another significant issue I want to fix.